src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CreditWaiter.cs

```diff
@@ -10,21 +10,28 @@
 namespace System.Net.Http
 {
     /// <summary>Represents a waiter for credit.</summary>
-    internal class CreditWaiter : IValueTaskSource<int>
+    internal sealed class CreditWaiter : IValueTaskSource<int>
     {
+        private CancellationTokenRegistration _registration;
+        private ManualResetValueTaskSourceCore<int> _source;
+        private int _completionReserved;
+
         public int Amount;
```

Comment:
Small suggestion: Instead of making Amount a public field, you could instead pass the Amount value to the constructor and to ResetForAwait, then make Amount a read-only property, to prevent accidental modification of this value.

Reply:
I'd like to leave it as-is. This state has no bearing on the implementation; it's really just additional data carried with the instance.